config/config.go

```diff
@@ -32,10 +33,13 @@ type Adapter struct {
 }
 
 type Metrics struct {
-	Host     string `mapstructure:"host"`
-	Database string `mapstructure:"database"`
-	Username string `mapstructure:"username"`
-	Password string `mapstructure:"password"`
+	Type     string        `mapstructure:"type"`
+	Host     string        `mapstructure:"host"`
+	Database string        `mapstructure:"database"`
+	Username string        `mapstructure:"username"`
+	Password string        `mapstructure:"password"`
+	Interval time.Duration `mapstructure:"interval"`
+	Prefix   string        `mapstructure:"prefix"`
```

Comment:
This might be a good time to make any breaking changes to be more flexible, before a bunch of people start relying on it.

One issue I'm seeing is that neither graphite nor statsd seem to recognize many of these fields. If so, then our config structure shouldn't allow them. It makes it clearer what's required (when writing the config) and what's valid (when reading values in the code).

I'm not sure what the limitations on viper's config & reading is... but is there some way we could do something like this?

```
metrics: {
  influx: {
    host: ...,
    database: ...,
  },
  graphite: {
    host: ...,
    interval: ...,
  }
}
```

That way we could define structs which only include fields that matter to the db in question.

Another (tiny) added benefit is that it makes the config API a bit more flexible so that we could export stats to several backends. Not sure if there's a use-case for that, though... so the main goal is definitely clarity.